.github/workflows/build.yml

```diff
@@ -0,0 +1,108 @@
+          echo "SHORT_SHA=${SHORT_SHA}" >> "$GITHUB_ENV"
+      - name: Setup variables (Windows)
+        if: ${{ startsWith(matrix.os, 'windows-') }}
+        run: |
+          echo "PYTHONHASHSEED=42" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+          $env:SHORT_SHA = "${{ github.sha }}".Substring(0, 7)
+          echo ("SHORT_SHA=" + $env:SHORT_SHA) | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+      - name: Install Linux build dependencies
+        if: ${{ startsWith(matrix.os, 'ubuntu-') }}
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r build_configs/linux/requirements.txt
+      - name: Install macOS build dependencies
+        if: ${{ startsWith(matrix.os, 'macos-') }}
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r build_configs/linux/requirements.txt
+      - name: Install Windows build dependencies
+        if: ${{ startsWith(matrix.os, 'windows-') }}
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r build_configs/windows/requirements.txt
+      - name: Build with build.spec (Linux amd64)
+        if: ${{ startsWith(matrix.os, 'ubuntu-') }}
+        run: |
+          export BUILD_FILE_NAME=staking_deposit-cli-${SHORT_SHA}-linux-amd64
+          echo "BUILD_FILE_NAME=${BUILD_FILE_NAME}" >> "$GITHUB_ENV"
+          mkdir ${BUILD_FILE_NAME}
+          pyinstaller --distpath ./${BUILD_FILE_NAME} ./build_configs/linux/build.spec
+      - name: Build with build.spec (macOS amd64)
+        if: ${{ matrix.os == 'macos-12' }}
+        run: |
+          export BUILD_FILE_NAME=staking_deposit-cli-${SHORT_SHA}-darwin-amd64
+          echo "BUILD_FILE_NAME=${BUILD_FILE_NAME}" >> "$GITHUB_ENV"
+          mkdir ${BUILD_FILE_NAME}
+          pyinstaller --distpath ./${BUILD_FILE_NAME} ./build_configs/macos/build.spec
+      - name: Build with build.spec (macOS arm64)
+        if: ${{ matrix.os == 'macos-latest' }}
+        run: |
+          export BUILD_FILE_NAME=staking_deposit-cli-${SHORT_SHA}-darwin-arm64
+          echo "BUILD_FILE_NAME=${BUILD_FILE_NAME}" >> "$GITHUB_ENV"
+          mkdir ${BUILD_FILE_NAME}
+          pyinstaller --distpath ./${BUILD_FILE_NAME} ./build_configs/macos/build.spec
+      - name: Build with build.spec (Windows amd64)
+        if: ${{ startsWith(matrix.os, 'windows-') }}
+        run: |
+          $env:BUILD_FILE_NAME = ("staking_deposit-cli-" + $env:SHORT_SHA + "-windows-amd64")
+          echo ("BUILD_FILE_NAME=" + $env:BUILD_FILE_NAME) | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+          mkdir $env:BUILD_FILE_NAME
+          $env:BUILD_FILE_NAME_PATH = (".\" + $env:BUILD_FILE_NAME)
+          echo ("BUILD_FILE_NAME_PATH=" + $env:BUILD_FILE_NAME_PATH) | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+          pyinstaller --distpath $env:BUILD_FILE_NAME .\build_configs\windows\build.spec
+      - name: Install coreutils (macOS)
```

Comment:
What is the purpose of this? I ran a build without this and tested against another version and saw no differences or am I just completely missing something here

https://github.com/valefar-on-discord/ethstaker-deposit-cli/actions/runs/8904716025/job/24454392596
vs
https://github.com/valefar-on-discord/ethstaker-deposit-cli/actions/runs/8904689777/job/24454331736

Reply:
coreutils provides the `sha256sum` binary on macOS since it's not installed by default

Reply:
Right you are. Thanks!